app/views/schedules/_task_fields.html.erb

```diff
@@ -19,11 +19,14 @@
                      class: 'form-control' %>
 
         <span class="input-group-btn">
-        <%= link_to_remove_association '',
-                                       f,
-                                       alt: 'Remove Task',
-                                       title: 'Remove Task',
-                                       class: 'btn btn-danger glyphicon glyphicon-minus' %>
+        <% unless !f.object.schedule.nil? and f.object.id == f.object.schedule.root_task_id %>
```

Comment:
That is just a clause so that the remove symbol shows up next to each task except the root task (so you can't accidentally delete the root). I can (and subsequently have) gotten rid of the first part of the unless clause checking if the schedule exists (it always will when task_fields is called).